pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -209,9 +215,35 @@ func (r *Builder) PodEnvironment(vmRef ref.Ref, sourceSecret *core.Secret) (env
 			Value: fingerprint,
 		},
 	)
+	if macsToIps != "" {
+		env = append(env, core.EnvVar{
+			Name:  "V2V_staticIPs",
+			Value: macsToIps,
+		})
+	}
 	return
 }
 
+func (r *Builder) mapMacStaticIps(vm *model.VM) string {
+	if !IsWindows(vm) {
+		return ""
+	}
+	configurations := []string{}
+	for _, guestNetwork := range vm.GuestNetworks {
+		if guestNetwork.Origin == string(types.NetIpConfigInfoIpAddressOriginManual) {
+			configurations = append(configurations, fmt.Sprintf("%s:ip:%s", guestNetwork.MAC, guestNetwork.IP))
+		}
+	}
+	return strings.Join(configurations, "_")
+}
+
+func IsWindows(vm *model.VM) bool {
+	if vm.GuestID != "" && strings.Contains(vm.GuestID, WindowsPrefix) || strings.Contains(vm.GuestName, WindowsPrefix) {
```

Comment:
why do we need to check both GuestID and GuestName here? in `TemplateLabels` we only check GuestName

Reply:
we check both with priority to `GuestID`.
```
if vm.GuestID != "" {
		os = osMap[vm.GuestID]
	} else if strings.Contains(vm.GuestName, "linux") || strings.Contains(vm.GuestName, "rhel") {
		os = DefaultLinux
	} else if strings.Contains(vm.GuestName, "win") {
		os = DefaultWindows
```

Reply:
yeah, ok, I missed the checks of GuestName below somehow